### PR TITLE
update ActorSystemSpec to use property based testing for its name

### PR DESCRIPTION
**Pull request by dongxuwang**

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/actor/ActorSystemSpec.scala [hunk 1]
```diff
@@ -115,23 +117,34 @@ object ActorSystemSpec {
...
+          "-hallowelt",
+          "_hallowelt",
+          "hallo welt",
+          "hallo*welt",
+          "hallo@welt",
+          "hallo#welt",
+          "hallo$welt",
+          "hallo%welt",
+          "hallo/welt")
+      forAll(commonInvalidNames) { (name: String) =>
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/actor/ActorSystemSpec.scala:138]: What is the value of using `forAll` in this test compared to how it was? If no value then revert.

Comment (dongxuwang) [reply akka-actor-tests/src/test/scala/akka/actor/ActorSystemSpec.scala:138]: Looks more human language, I think.

Comment (patriknw) [reply akka-actor-tests/src/test/scala/akka/actor/ActorSystemSpec.scala:138]: For us, plain Scala is more natural language. 😄 
Let's revert this part.

Comment (dongxuwang) [reply akka-actor-tests/src/test/scala/akka/actor/ActorSystemSpec.scala]: updated, could you help take a look

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**